QCircLearning/utils.py

```diff
@@ -64,3 +64,15 @@ def __call__(self, current_loss):
             if self.counter >= self.patience:
                 return True
         return False
+
+
+def reinitialize_network(model):
+    """
+    Re-initializes the parameters of a PyTorch network in place
+    (without constructing a new model).
+
+    model (nn.Module): The model to re-init.
+    """
+    for module in model.modules():
+        if hasattr(module, "reset_parameters"):
+            module.reset_parameters()
```

Comment:
Why we need to check the attribute `reset_parameters` here? In what case a `module` may not define the `reset_parameters`?

Reply:
model.modules() will contains activation function or other non-parametric modules that does not have reset_parameters